File: Python/DAY_1/menuDriven.py
def is_palindrome(n):
    return str(n) == str(n)[::-1]

def is_prime(n):
    if n < 2:
        return False
    for i in range(2, int(n**0.5)+1):
        if n % i == 0:
            return False
    return True

def is_perfect(n):
    sum_div = 0
    for i in range(1, n):
        if n % i == 0:
            sum_div += i
    return sum_div == n

while True:
    print("\nMenu:")
    print("1. Check Palindrome")
    print("2. Check Prime Number")
    print("3. Check Perfect Number")
    print("4. Exit")

    choice = input("Enter your choice (1-4): ")

    if choice == "1":
        num = int(input("Enter a number to check palindrome: "))
        if is_palindrome(num):
            print(f"{num} is a Palindrome")
        else:
            print(f"{num} is NOT a Palindrome")

    elif choice == "2":
        num = int(input("Enter a number to check prime: "))
        if is_prime(num):
            print(f"{num} is a Prime Number")
        else:
            print(f"{num} is NOT a Prime Number")

    elif choice == "3":
        num = int(input("Enter a number to check perfect number: "))
        if is_perfect(num):
            print(f"{num} is a Perfect Number")
        else:
            print(f"{num} is NOT a Perfect Number")

    elif choice == "4":
        print("Exiting program. Goodbye!")
        break

    else:
        print("Invalid choice. Please enter 1-4.")

File: C/DAY_1/menuDriven.c
#include <stdio.h>

int main() {
    while (1) {

        printf("\n Press 1 ➤ Check prime within range");
        printf("\n Press 2 ➤ Check perfect within range");
        printf("\n Press 3 ➤ Check palindrome within range");
        printf("\n Press 4 ➤ Print Diamond");
        printf("\n Press 5 ➤ Exit");

        int choice;
        printf("\n\nEnter choice = ");
        scanf("%d", &choice);

        if (choice == 5) {
            printf("\nThank you!! Goodbye!!\n");
            break;
        }

        int low, high, count = 0;

        if (choice == 1) {
            printf("Enter low and high = ");
            scanf("%d %d", &low, &high);

            for (int number = low; number <= high; number++) {
                int isPrime = 1; 
                if (number < 2) isPrime = 0; 
                for (int i = 2; i <= number / 2; i++) {
                    if (number % i == 0) {
                        isPrime = 0;
                        break;
                    }
                }
                if (isPrime == 1) {
                    printf("%d ", number);
                    count++;
                }
            }
            printf("\nTotal primes = %d\n", count);
        } 
        else if (choice == 2) {
            printf("Enter low and high = ");
            scanf("%d %d", &low, &high);

            for (int number = low; number <= high; number++) {
                int sumOfDivisors = 0;
                for (int i = 1; i <= number / 2; i++) {
                    if (number % i == 0) {
                        sumOfDivisors += i;
                    }
                }
                if (sumOfDivisors == number) {
                    printf("%d ", number);
                    count++;
                }
            }
            printf("\nTotal perfect numbers = %d\n", count);
        } 
        else if (choice == 3) {
            printf("Enter low and high = ");
            scanf("%d %d", &low, &high);

            for (int number = low; number <= high; number++) {
                int original = number;
                int reverse = 0;

                while (original != 0) {
                    int digit = original % 10;
                    reverse = reverse * 10 + digit;
                    original /= 10;
                }

                if (reverse == number) {
                    printf("%d ", number);
                    count++;
                }
            }
            printf("\nTotal palindromes = %d\n", count);
        } 
       
        else {
            printf("\nWrong input\n");
        }
    }

    return 0;
}
